siege_utilities/schema/tests/test_migration_runner_integration.py

```
@@ -0,0 +1,213 @@
+        runner.apply_up_to("0099")
+
+
+def test_failed_migration_rolls_back_and_is_not_recorded(
+    pg_dsn, migrations_dir, tracking_schema
+):
+    _write(migrations_dir, "V0001__good.sql", "SELECT 1;")
+    _write(migrations_dir, "V0002__bad.sql", "SELECT * FROM does_not_exist;")
+    _write(migrations_dir, "V0003__should_not_run.sql", "SELECT 1;")
+    runner = _runner(pg_dsn, migrations_dir, tracking_schema)
+
+    with pytest.raises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)